src/plugins/data/common/datasets/types.ts

```diff
@@ -125,7 +142,12 @@ export interface DataStructure {
   children?: DataStructure[];
   hasNext?: boolean;
   paginationToken?: string;
+  /** Whether data structure supports multi-selecting children. The selected
+   * children will be combined using {@link DatasetTypeConfig['combineDataStructures']} */
   multiSelect?: boolean;
+  /** Custom component to select children and create next data structure or
+   * dataset. multiSelect will be ignored if using custom component */
+  DataStructureCreator?: React.ComponentType<DataStructureCreatorProps>;
```

Comment:
nit:

would it make more sense to be like
`customExplorerComponent` or `customComponent` or something. contract wise to put the comment into the name.`

Reply:
wouldn't they be too generic? Developers probably won't understand what the component does by looking at the name